library/src/hipsolver_module.f90

```diff
@@ -62,41 +62,41 @@ module hipsolver
     
     interface
         function hipsolverCreate(handle) &
-                result(c_int) &
                 bind(c, name = 'hipsolverCreate')
             use iso_c_binding
             implicit none
+            integer(kind(HIPSOLVER_STATUS_SUCCESS)) :: hipsolverCreate_
```

Comment:
Have you tried to compile the F90 code with other compilers such as gfortran, just to check the syntax? For example, on line 68, there is reference to HIPSOLVER_STATUS_SUCCESS in another module hipsolver_enums. Perhaps there should be "use hipsolver_enums"? 

The "result" names the variable so we need another line to declare the type of this result variable.